docs/source/extensions/extension_development.rst

```diff
@@ -0,0 +1,212 @@
+``simulate``).
+
+Extensions in ASReview are Python packages and can extend the
+subcommands of asreview (see ``asreview -h``).
+
+The easiest way to create a new subcommand is by defining a class that can be
+used as a new entry point for ASReview. This class should inherit from
+:class:`asreview.entry_points.base.BaseEntryPoint`. Add the functionality to the
+class method ``execute``.
+
+.. code:: python
+
+    from asreview.entry_points import BaseEntryPoint
+
+    class ExampleEntryPoint(BaseEntryPoint):
+
+        description = "Description of example extension"
+        extension_name = "asreview-example"  # Name of the extension
+        version = "1.0"  # Version of the extension in x.y(.z) format.
+
+        def execute(self, argv):
+            pass  # Implement your functionality here.
+
+It is strongly recommended to define the attributes ``description``,
+``extension_name``, and ``version``.
+
+The class method ``execute`` accepts a positional arugument (``argv`` in this
+example).  First create the functionality you would like to be able to use in
+any directory. The argument ``argv`` are the command line arguments left after
+removing asreview and the entry point.
+
+It is advised to place the newly defined class ``ExampleEntryPoints`` in the
+following package structure:
+``asreviewcontrib.{extension_name}.{your_modules}``. For example:
+
+.. code:: bash
+
+    ├── README.md
+    ├── asreviewcontrib
+    │   └── example
+    │       ├── __init__.py
+    │       ├── entrypoint.py
+    │       └── example_utils.py
+    ├── setup.py
+    └── tests
+
+
+Create a ``setup.py`` in
+the root of the package, and set the keyword argument `entry_points` of
+``setup()`` under ``asreview.entry_points``, for example:
+
+.. code:: python
+
+    entry_points={
+        "asreview.entry_points": [
+            "example = asreviewcontrib.example.entrypoint:ExampleEntryPoint",
+        ]
+    }
+
+After installing this package, ASReview is extended with the ``asreview
+example`` subcommand. See ``asreview -h`` for this option.
+
+An example of a subcommand extension is the `Visualization Extension
+<https://github.com/asreview/asreview-visualization>`_
+
+
+
+
+
+
+
+
+
+
+
+
+Dataset extensions
+------------------
+An extension of the :class:`asreview.datasets.BaseDataSet` class.
+
+Dataset extensions integrate new datasets for use in ASReview. Adding datasets
+via extension provides quick access to the dataset via commandline or other user
+interface.
+
+It is advised to place the new dataset ``your_dataset`` in the
+following package structure:
+
+.. code:: bash
+
+    ├── README.md
+    ├── asreviewcontrib
+    │   └── dataset_name
+    │       ├── __init__.py
+    │       └── your_dataset.py
+    ├── config
+    │   └── your_dataset.json
+    ├── datasets
+    │   └── dataset_name
+    │       ├── __init__.py
+    │       └── your_dataset.csv
+    ├── setup.py
+    └── tests
+
```

Comment:
I think we need an example of `your_dataset.py` here. Covid plugin is relatively complicated.

I think a minimal example would look like this:

```
from asreview.datasets import BaseDataSet
from asreview.datasets import BaseDataGroup

class YourDataGroup(BaseDataGroup):
    group_id = "your_data_group"
    description = "A new data group with my awesome datasets."

    def __init__(self):
        
        dataset = BaseDataSet.from_config({
            "dataset_id": "your_data_id",
            "url": "",
            "reference": "",
            "link": "",
            "license": "",
            "title": "Your Data",
            "authors": [
              "Jane Doe",
              "John Doe"
            ],
            "year": 2021,
            "topic": "Your topic",
            "final_inclusions": true,
            "title_abstract_inclusions": false
          }
        )

        super(YourDataGroup, self).__init__(dataset)
        # pass multiple datasets to init if there are more datasets
```